xds/internal/balancer/clusterresolver/e2e_test/balancer_test.go

```diff
@@ -302,3 +313,217 @@ func (s) TestErrorFromParentLB_ResourceNotFound(t *testing.T) {
 		t.Fatalf("RPCs did not fail after removal of Cluster resource")
 	}
 }
+
+// testChildPolicyBalancerBuilder wraps the priority LB policy (which is the
+// child policy for the cluster resolver LB policy), and makes certain events
+// available to the test (subConn state changes and config pushes).
+type testChildPolicyBalancerBuilder struct {
```

Comment:
Nit: this name is really long, and I'd prefer calling it something with "priority", rather than "child", since it does know it's specific child in the xDS specific tree. Can we shorten this builders name and also the balancers name. Maybe something like wrappedPriority/wrappedPriorityBuilder.